tests/ci/test_optional_dep_guards.py

```
@@ -174,7 +175,28 @@ class bodies are already guarded by their enclosing scope.
-    """Return test files modified relative to main (diff-based subset)."""
+    """Return test files modified relative to main (diff-based subset).
+
+    In CI, a shallow checkout can leave ``origin/main`` unresolvable, which
+    would otherwise silently collapse the diff to empty and skip every
+    violation. Fail loudly in that case rather than degrade to a no-op
+    guardrail. Outside CI (local dev, possibly without a configured
+    ``origin`` remote) the silent fallback below is intentional.
+    """
+    if os.environ.get("GITHUB_ACTIONS") == "true":
+        verify = subprocess.run(
+            ["git", "rev-parse", "--verify", "--quiet", "origin/main"],
```

Ruff (0.15.18)

[error] Starting a process with a partial executable path

(S607)